Fix Opal on a C++ client

theiostream opened the pull request:
These two commits allow Opal to work on a C++ context.

theiostream committed:
headers: remove typedef declarations for enums

Remove `typedef enum` declarations from public Opal headers, which
caused type errors if a client tried to bitwise-OR two values of
different enum types, which is usually allowed per the API
specification.

theiostream committed:
CGPath: export `CGMutablePathRef` as `CGPath *`

Export the `CGMutablePathRef` as `CGPath *`, and not `CGMutablePath *`.
This allows CGPathRef to be replaced with CGMutablePathRef seamlessly on
a C++ context, improving compatibility with Quartz.

theiostream committed:
headers: allow Opal to be used from within C++

Add `extern "C"` declarations to all Opal headers to allow them to be
used from within C++ programs.

theiostream commented:
@ivucica

ivucica commented:
Second commit makes sense, for the first one I'm roping in @ericwa again

theiostream commented:
Just a brief comment then: I am far from being certain that my solution to this issue is the best; but in C++, `CGPathRef x = CGPathCreateMutable();` would error; which is not API-compatible. This needs to be addressed _somehow_.

ericwa commented on Headers/CoreGraphics/CGPath.h:
```diff
@@ -42,16 +42,11 @@ typedef enum CGPathElementType {
-#else
-typedef struct CGMutablePath* CGMutablePathRef;
+typedef const struct CGPath* CGPathRef;
+typedef struct CGPath* CGMutablePathRef;
```
Couldn't we keep the CGMutablePath/CGPath classes as-is internally, but still make the public API change, i.e. this part:?
```
#ifdef INTERNAL_BUILD_OBJC
...
#else
typedef const struct CGPath* CGPathRef;
typedef struct CGPath* CGMutablePathRef;
#endif
```

theiostream replied:
Theoretically, yes. Stupid of me not to think of it. I'll see if anything goes wrong with that approach.

theiostream committed:
CGGeometry: define CG{Point,Size,Rect} as NS{Point,Size,Rect}

Define CGGeometry basic structs as equivalent to their NSGeometry
counterparts. This allows them to be interchangeable as in the reference
platform.

theiostream commented:
@ericwa @ivucica I grouped some more header API compatibility fixes in this pull request. Could you comment?

Thank you for the feedback so far.

ericwa commented:
Looks good to me.

```
enum {
...
} CTFontSymbolicTraits;
typedef int CTFontSymbolicTraits;
```
seems to be the standard way enums are done in libs-gui.

theiostream commented:
I suppose you meant:

```
enum {
...
};
```

, without the last `CTFontSymbolicTraits`, which is what this pull request does? Otherwise you're declaring a global variable called `CTFontSymbolicTraits`.

ericwa commented:
Sorry, typo on my part. I just meant that what you have in the pull request looks good.

theiostream commented:
Thanks. Waiting for final feedback from @ivucica then :)

fredkiefer reviewed (commented):
Looks good to me.

theiostream committed:
Update ChangeLog

ivucica commented:
I see no problems upon a (quick) inspection. Merging.
